benchmarks/benchmarks/optimize.py

```diff
@@ -262,6 +264,48 @@ def bench_run(self, x0, methods=None, **minimizer_kwargs):
                 self.add_result(res, t1-t0, method)
 
 
+class BenchDFO(Benchmark):
+    """
+    Benchmark the optimizers with the CUTEST DFO benchmark of Moré and Wild.
+    The original benchmark suite is available at https://www.mcs.anl.gov/~more/dfo/.
+    """
+    params = [
+        [0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,
+        20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,
+        40,41,42,43,44,45,46,47,48,49,50,51,52],
+        ["COBYLA", 'Powell', 'nelder-mead',
+         'L-BFGS-B', 'BFGS', 'trust-constr'],
+        ["mean_nfev", "mean_time"]
+    ]
+    param_names = ["DFO benchmark problem number", "solver", "result type"]
+
+    def setup(self, prob_number, method_name, ret_val):
```

Comment:
Not really related to this PR since I see that the rest of this module is built the same way as well, but this is certainly quite different from the normal structure of conventional `asv` benchmarks.

I assume there's a good reason for having a custom `bench_run()` throughout this file. Out of curiosity, does `asv continuous master my_branch` even work for assessing performance improvements in `optimize`? The PR description here goes through a different mechanism to compare for example.

Reply:
This is my first time working with `asv`, so I modeled my benchmark on the rest of the module. However, when I try to compare branches (to benchmark this PR) with `asv` it gets stuck and times out creating the environments. So, I ended up just making my own local branches with the old and new versions, running the benchmark in each branch, and manually comparing the results.